common/lib/xmodule/xmodule/tests/test_video.py

```diff
@@ -522,21 +536,75 @@ def test_import_with_float_times(self):
             'data': ''
         })
 
+    @patch('xmodule.video_module.video_module.edxval_api')
+    def test_import_val_data(self, mock_val_api):
+        def mock_val_import(xml, edx_video_id):
+            """Mock edxval.api.import_from_xml"""
+            self.assertEqual(xml.tag, 'video_asset')
+            self.assertEqual(dict(xml.items()), {'mock_attr': ''})
+            self.assertEqual(edx_video_id, 'test_edx_video_id')
+
+        mock_val_api.import_from_xml = Mock(wraps=mock_val_import)
+        module_system = DummySystem(load_error_modules=True)
+
+        # import new edx_video_id
+        xml_data = """
+            <video edx_video_id="test_edx_video_id">
+                <video_asset mock_attr=""/>
+            </video>
+        """
+        video = VideoDescriptor.from_xml(xml_data, module_system, Mock())
+
+        self.assert_attributes_equal(video, {'edx_video_id': 'test_edx_video_id'})
+        mock_val_api.import_from_xml.assert_called_once_with(ANY, 'test_edx_video_id')
+
+    @patch('xmodule.video_module.video_module.edxval_api')
+    def test_import_val_data_invalid(self, mock_val_api):
+        mock_val_api.ValCannotCreateError = _MockValCannotCreateError
+        mock_val_api.import_from_xml = Mock(side_effect=mock_val_api.ValCannotCreateError)
+        module_system = DummySystem(load_error_modules=True)
+
+        xml_data = """
+            <video edx_video_id="test_edx_video_id">
+                <video_asset client_video_id="test_client_video_id" duration="-1"/>
+            </video>
+        """
+        with self.assertRaises(mock_val_api.ValCannotCreateError):
+            VideoDescriptor.from_xml(xml_data, module_system, Mock())
```

Comment:
Is it the `-1` duration that causes the assert? If so, are there other validations that are performed? What happens when invalid attrs are in the XML? Are they just ignored?